presto-elasticsearch/src/main/java/com/facebook/presto/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -0,0 +1,254 @@
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.ExistsQueryBuilder;
+import org.elasticsearch.index.query.MatchAllQueryBuilder;
+import org.elasticsearch.index.query.MatchQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.index.query.TermQueryBuilder;
+import org.elasticsearch.transport.client.PreBuiltTransportClient;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.facebook.presto.elasticsearch.ElasticsearchErrorCode.ELASTIC_SEARCH_CLIENT_ERROR;
+import static com.facebook.presto.spi.predicate.Marker.Bound.ABOVE;
+import static com.facebook.presto.spi.predicate.Marker.Bound.BELOW;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.type.IntegerType.INTEGER;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Iterables.getOnlyElement;
+import static java.util.Objects.requireNonNull;
+
+public class ElasticsearchQueryBuilder
+{
+    private static final Logger log = Logger.get(ElasticsearchQueryBuilder.class);
+
+    private final Duration scrollTime;
+    private final int scrollSize;
+    private final Client client;
+    private final int shard;
+    private final TupleDomain<ColumnHandle> tupleDomain;
+    private final List<ElasticsearchColumnHandle> columns;
+    private final String index;
+    private final String type;
+
+    public ElasticsearchQueryBuilder(List<ElasticsearchColumnHandle> columnHandles, ElasticsearchConnectorConfig config, ElasticsearchSplit split)
+    {
+        columns = requireNonNull(columnHandles, "ElasticsearchColumnHandle is null");
+        requireNonNull(config, "ElasticsearchConnectorConfig is null");
+        requireNonNull(split, "ElasticsearchSplit is null");
+        ElasticsearchTableDescription table = split.getTable();
+        tupleDomain = split.getTupleDomain();
+        index = table.getIndex();
+        type = table.getType();
+        shard = split.getShard();
+        try {
+            Settings settings = Settings.builder().put("cluster.name", table.getClusterName()).build();
+            client = new PreBuiltTransportClient(settings).addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(split.getSearchNode()), table.getPort()));
+        }
+        catch (IOException e) {
```

Comment:
This is the only place `ELASTIC_SEARCH_CLIENT_ERROR` is used and the usage is confusing (based on the name of this error code).

The only exception thrown from this block of code is an `UnknownHostException` from `InetAddress`, so it's confusing here that we are catching `IOException` *and*  that we are throwing an `ELASTIC_SEARCH_CLIENT_ERROR`.

The name of this error code sounds like it's an error in the elastic search client, but here we throw it when there is an address resolution issue when creating the client.